pkg/router/routeapihelpers/validation.go

```diff
@@ -205,6 +205,11 @@ func ExtendedValidateRoute(route *routev1.Route) field.ErrorList {
 			} else {
 				tlsConfig.CACertificate = string(data)
 			}
+			// HAProxy will fail to start if intermediate CA certs use unsupported signature algorithms.
+			// However, root CAs can still use unsupported algorithms since they are self-signed.
```

Comment:
Should you only check the CACertificate for whether it's self-signed, instead of within `validateCertSignatureAlgorithms`?  In other words, is a self-signed cert ok for a `DestinationCACertificate`?